halo2_proofs/src/poly/query.rs

```diff
@@ -79,22 +93,31 @@ impl<'com, C: CurveAffine, M: MSM<C>> VerifierQuery<'com, C, M> {
 }
 
 /// A polynomial query at a point
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct VerifierQuery<'com, C: CurveAffine, M: MSM<C>> {
-    /// point at which polynomial is queried
+    /// Point at which polynomial is queried
     pub(crate) point: C::Scalar,
-    /// commitment to polynomial
+    /// Commitment to polynomial
     pub(crate) commitment: CommitmentReference<'com, C, M>,
-    /// evaluation of polynomial at query point
+    /// Evaluation of polynomial at query point
     pub(crate) eval: C::Scalar,
 }
 
-impl<'com, C: CurveAffine, M: MSM<C>> Clone for VerifierQuery<'com, C, M> {
-    fn clone(&self) -> Self {
-        Self {
-            point: self.point,
-            commitment: self.commitment,
-            eval: self.eval,
+impl<'com, C, M> VerifierQuery<'com, C, M>
+where
+    C: CurveAffine,
+    M: MSM<C>,
+{
+    /// Create a new verifier query based on a commitment
+    pub fn new(
```

Comment:
ditto.

Reply:
Just assumed that we might need this for further use. For now `VerifierQuery::new_commitment()` would be enough.